### ISSUE TITLE
Meta-metadata: allow specification of metadata schema according to standards

### ISSUE THREAD

**Issue by emdalton** [PoetOS/moodle-local_metadata#6]
The ability to identify any given metadata field as a part of a standard could be very helpful in making the fields more generally usable, e.g. by other plugins. It could work by adding some options when a new field is created in a context: at the same time that the field is defined, give the option of selecting a supported schema and choosing a metadata element to identify the field with. Additional schemae could be supported by plugins.

For those not familiar with these metadata standards, here's a fairly readable introduction: http://publications.cetis.org.uk/wp-content/uploads/2011/02/WhatIsIEEELOM.pdf

As a key use case, this could allow new optional fields (metadata fields) to be included in Learning Analytics calculations if their purpose was identified, and different sites would be free to implement the fields independently to meet local needs, while still describing the data well enough to be included in a general-purpose Learning Analytics algorithm.

How this would look:

1 - At the site level, when enabling metadata for a context, the admin can also select "enable metadata schema" and/or "restrict metadata to schema" and select a schema.

2 - When adding a field, if either "enable" or "restrict" schema options are on, present a combo box control to allow the metadata element to be selected when the field is created. If "restrict," this is a mandatory field.

I suggest the IMS LRM/IEEE LOM as the first supported schema: https://www.imsglobal.org/metadata/index.html

Additional schemae (e.g. Dublin Core) could be supported in the future by using plugins.

I suggest this modification because a little bit of advanced planning at this stage could make this feature much more useful in the future.

**Comment by mchurchward** [comment]
Definitely. This was one of the key ideas at the start of this. I'll look into any low-level changes that would need to be done to support this. I think it may simply need one more level of "category" that includes "schema".
